src/SnackHub.Domain/Entities/KitchenOrder.cs

```diff
@@ -29,13 +29,20 @@ public KitchenOrder(Guid id, Guid orderId, IReadOnlyCollection<KitchenOrdertItem
 
     public void UpdateStatus()
     {
+        var previousStatus = Status;
+        
         Status = Status switch
         {
             KitchenOrderStatus.Received => KitchenOrderStatus.Preparing,
-            KitchenOrderStatus.Preparing => KitchenOrderStatus.Finished,
-            KitchenOrderStatus.Finished => KitchenOrderStatus.Done,
+            KitchenOrderStatus.Preparing => KitchenOrderStatus.Done,
+            KitchenOrderStatus.Done => KitchenOrderStatus.Finished,
```

Comment:
⚠️ check with @IcaroSousa if this makes sense: transitions were updated to match the enum order / descriptions